tests/cli/test_cmds_check.py

```diff
@@ -285,6 +287,51 @@ def test_check_smartstack_check_is_ok_when_no_smartstack(mock_is_file_in_dir, ca
     assert output == expected_output
 
 
+@patch("paasta_tools.cli.cmds.validate.get_config_file_dict", autospec=True)
+def test_check_parallel_works(mock_get_config_file_dict):
+    # Parallel steps is seen as valid
+    mock_get_config_file_dict.return_value = {
+        'pipeline': [{
+            'parallel': [
+                {
+                    'step': 'somecluster.something',
+                },
+                {
+                    'step': 'somesecurity.check',
+                },
+            ],
+            'step': 'anothercluster.something',
+            'wait_for_deployment': False,
+            'timeout': 3600,
+        }],
+    }
+
+    actual = validate_schema('my/fake-service','deploy') is True
+    assert actual is True
+
+@patch("paasta_tools.cli.cmds.validate.get_config_file_dict", autospec=True)
+def test_check_parallel_fails(mock_get_config_file_dict):
+    # Parallel steps is seen as valid
+    mock_get_config_file_dict.return_value = {
+        'pipeline': [{
+            'invalid-prop': [
+                {
+                    'step': 'somecluster.something',
+                },
+                {
+                    'step': 'somesecurity.check',
+                },
+            ],
+            'step': 'anothercluster.something',
+            'wait_for_deployment': False,
+            'timeout': 3600,
+        }],
+    }
```

Comment:
if we wanted to get real fancy, we could do something like this to make this test a little more modern (compared to the others in this file) by using pytest parametrization + mocking with a context manager:
(note - i wrote this in the github text field so the formatting is all messed up :p)
```suggestion
@pytest.mark.parametrize("config_dict,is_valid", (
        ({
        'pipeline': [{
             'parallel': [
                 {
                     'step': 'somecluster.something',
                 },
                 {
                     'step': 'somesecurity.check',
                },
             ],
             'step': 'anothercluster.something',
             'wait_for_deployment': False,
             'timeout': 3600,
          }],
    }, True),
    # invalid config - not expecting another key to have steps nested inside
    ({
        'pipeline': [{
             'not-parallel': [
                 {
                     'step': 'somecluster.something',
                 },
                 {
                     'step': 'somesecurity.check',
                },
             ],
             'step': 'anothercluster.something',
             'wait_for_deployment': False,
             'timeout': 3600,
          }],
    }, False),
  )
)
def test_check_parallel_schema(config_dict, is_valid):
    # Parallel steps is seen as valid
    with mock.patch("paasta_tools.cli.cmds.validate.get_config_file_dict", autospec=True, return_value=config_dict):
        assert validate_schema('my/fake-service','deploy') is is_valid
```
but this is just a suggestion - feel free to merge without doing this

Reply:
I actually like this fanciness. The only slight change is the return type for the failure would actually be None